Add ability for set::static-quit to only block unregistered users

dboyz opened the pull request:
TODO: Update documentation files.

dboyz committed:
Add ability for set::static-quit to only block unregistered users

TODO: Update documentation files.

dboyz committed:
Oops

tmcarthur commented:
Having a look.

dboyz commented:
Thanks! :)

tmcarthur commented:
No problem - let me know if you need any assistance.

tmcarthur commented:
Also - you still planning on updating the pull request with your changes :)? No worries either way, just want to be sure.

dboyz commented:
Yeah. I'm still figuring out things sorry.

tmcarthur commented:
No problem or rush - just wanted to be sure :)

dboyz committed:
Remove anti-spam-quit-message-time part 1

dboyz committed:
Remove anti-spam-quit-message-time part 2

dboyz committed:
Introduce static_quit_part_set and static_quit_part_time

dboyz committed:
Introduce static_quit_part_set and static_quit_part_time part 2

dboyz committed:
Introduce missing ircfree(i->static_part)

dboyz committed:
Removed anti-spam-quit-message-time, Introduce static_quit_part_set and static_quit_part_time part 3

dboyz committed:
Introduce static_quit_part_set and static_quit_part_time part 4

dboyz committed:
Make STATIC_QUIT actually appear.

dboyz committed:
Removed anti-spam-quit-message-time, Introduce static_quit_part_set and static_quit_part_time part 5

dboyz committed:
Merge pull request #1 from dboyz/unrealircdpatch2

Tweaked STATIC_QUIT and STATIC_PART and integrates with previously known ANTI_SPAM_QUIT_MESSAGE_TIME

dboyz commented:
I expect some errors in my pull request as this is my first time. Kindly take a look. Thank you.

tmcarthur commented:
Thanks, I will review today :)

dboyz commented:
I think moving forward for 3.4, this feature should be separated into a module. There are many stuff that can/should be separated into a module.

tmcarthur commented:
Agreed, but all of your changes except config parser are in a module, so I'm not particularly concerned with this change in terms of diminishing modularity.  I'll have a look at the code though and let you know if I have any concerns :).  Thanks again!

dboyz commented:
Thanks :)

tmcarthur commented:
Have a few concerns, but nothing major, looks okay.  Will add more notes shortly, but mostly around making sure no changes to config file are necessary if people want identical behavior to in previous versions :).  Thanks.

dboyz commented:
Alright. We may want to discuss on that a little further.

tmcarthur commented on src/s_conf.c:
```diff
@@ -7523,10 +7556,6 @@ int	_test_set(ConfigFile *conf, ConfigEntry *ce)
 				}
 			}
 		}
-		else if (!strcmp(cep->ce_varname, "anti-spam-quit-message-time")) {
```
I think we should should keep old option available so people who like current functionality don't have to change config.

tmcarthur commented on src/s_conf.c:
```diff
@@ -6839,6 +6840,33 @@ int	_conf_set(ConfigFile *conf, ConfigEntry *ce)
+		 * unregistered users only -dboyz
+		 * TODO: Produce error message by adding else
+		 */
+		else if (!strcmp(cep->ce_varname, "static-quit-part-set")) {
```
how about static-quit-part-users instead, since we're not "setting" it, we're setting which users it impacts?

tmcarthur commented on src/modules/m_quit.c:
```diff
@@ -91,7 +91,24 @@ DLLFUNC int  m_quit(aClient *cptr, aClient *sptr, int parc, char *parv[])
+			 */
+			if (STATIC_QUIT_PART_TIME == -1)
+			{
+				if!(IsLoggedIn(sptr) && STATIC_QUIT_PART_SET)
```
Does this compile? if! should not work here.  You need if(!IsLoggedIn....

tmcarthur commented on src/modules/m_quit.c:
```diff
@@ -91,7 +91,24 @@ DLLFUNC int  m_quit(aClient *cptr, aClient *sptr, int parc, char *parv[])
+				if!(IsLoggedIn(sptr) && STATIC_QUIT_PART_SET)
+					return exit_client(cptr, sptr, sptr, STATIC_QUIT);
+			}
+			else
```
Please use brackets with this else (so else { .... } ) since its next statement spans multiple lines

tmcarthur commented on src/modules/m_part.c:
```diff
@@ -101,12 +101,16 @@ DLLFUNC CMD_FUNC(m_part)
+				if!(IsLoggedIn(sptr) && STATIC_QUIT_PART_SET)
+					commentx = STATIC_PART;
+			}
 			else
```
Same comment on the else

tmcarthur commented on src/modules/m_part.c:
```diff
@@ -101,12 +101,16 @@ DLLFUNC CMD_FUNC(m_part)
-				commentx = STATIC_PART;
+				/* we only care of users within the time range -dboyz */
+				if (sptr->firsttime+STATIC_QUIT_PART_TIME > TStime())
+					if!(IsLoggedIn(sptr) && STATIC_QUIT_PART_SET)
```
Same comment on the if

tmcarthur commented:
So made my major comments - have some concerns around us not breaking existing behavior as well that are a little subtle.  So, from here, I'm happy to make some changes and merge this in, or you can make the changes if you'd like - I don't know if you're more interested in the feature getting in and tired of going back and forth, or if you want to see it through to completion on your own - just let me know, I'm happy doing either :)

dboyz commented:
I'll have a look at your comments. Will get back on this ASAP. I don't mind making the changes, it's part of my learning process haha.

dboyz committed:
Resolving conflict (1)

(Temporarily?) reintroduce anti_spam_quit_message_time.

dboyz committed:
Merge pull request #2 from unrealircd/unreal34

Fix build with LibreSSL

dboyz committed:
Renaming to static_quit_part_users (1)

dboyz committed:
Syntax error fix and etc.

Fix silly mistakes (i.e. missing braces and parentheses), 
Revert to ANTI_SPAM_QUIT_MSG_TIME, 
Switch to STATIC_QUIT_PART_USER

dboyz committed:
Reintroduce ANTI_SPAM_QUIT_MSG_TIME (2)

And introduce braces. Is this necessary?

dboyz committed:
Update comment and etc.

Forgot to update existing ANTI_SPAM_QUIT_MSG_TIME code.

dboyz committed:
Improve condition checking and compability with 3.2

dboyz committed:
Syntax error fix and etc. (2)

Fix silly mistakes (i.e. missing braces and parentheses), 
Revert to ANTI_SPAM_QUIT_MSG_TIME, 
Switch to STATIC_QUIT_PART_USER, 
Further enhance configuration flexibility.

dboyz committed:
Reintroduce ANTI_SPAM_QUIT_MSG_TIME (4), switch to STATIC_QUIT_PART_USER, update comments.

dboyz committed:
Switch to STATIC_QUIT_PART_USER (5)

dboyz committed:
Reintroduce ANTI_SPAM_QUIT_MSG_TIME (5), switch to STATIC_QUIT_PART_USERS (6), simplify codes.

dboyz committed:
Indention fix and another small correction

dboyz committed:
Stupid indentation fix

dboyz committed:
Knock out last occuring STATIC_QUIT_PART_TIME

dboyz commented:
Sorry for the late update. 26 commits for a simple pull request. Hopefully that number doesn't add up anymore.

tmcarthur commented:
No problem. I will have a look


> On Apr 21, 2015, at 2:29 AM, dboyz <notifications@github.com> wrote:
> 
> Sorry for the late update. 26 commits for a simple pull request. Hopefully that number doesn't add up anymore.
> 
> —
> Reply to this email directly or view it on GitHub.
>

dboyz commented:
Thanks a lot! I have a few questions and perhaps would be great if I could contact you elsewhere too (i.e. IRC, email, etc.?)

tmcarthur commented:
Sure - heero@unrealircd.com - I generally idle on the dev channel as well - nick is Heero

syzop commented:
So.. this pull request has like 20 patches associated with it. What do we do with it?

dboyz commented:
I'll re-work on this. Let's start with `git squash` :P
